### ISSUE TITLE
Character sheet show blank page

### ISSUE THREAD

**Issue by sioc-de-narf** [Rolisteam/rolisteam#108]
#Your configuration:
Rolisteam version: 
Operating System and version :  Windows 10

Rolisteam Setting (remove irrelevant):  
* GM and Server (localhost)
* Client 

#What did you do ? (Explain each step from starting rolisteam to the issue)
- Loaded a previously saved .rcs file
- shared a sheet with my client 

#What did you expect to see ? 
- The characters sheets as they are displayed in 1.8.2

#What did you get instead ? (Describe in details the issue) 
- A blank page on both GM and Server AND and on client only

I tried on many files even some I shared on GitHub, same result
And if I regenerate the code from RCSE, i lost background images...

Is there something to do to upgrade character sheet from 1.8.2 to 1.9 ?

**Comment by obiwankennedy** [comment]
yes Compatibility breaks between both version. I plan to make a webpage to convert old sheet to new standard.

**Comment by sioc-de-narf** [comment]
It took me hours to make those sheets... 
Please do that very fast, releasing without retrocompatibility or upgrade ability is just a non-sense. **And that should be said in the release notes !**
I won't use anymore that release before we can use the previous version sheets, i use them too much now, and i won't redo them.

**Comment by obiwankennedy** [comment]
Please read the whole answer. I describe many things to answer the issue.

It always like this: 
1.6 is not retrocompatible with 1.7
1.7 is not retrocompatible with 1.8
1.8 is not retrocompatible with 1.9.

this is why all previous version of rolisteam can still be downloaded.
One day I will bring retrocompatibility, but one day. 
Before it wasn't an issue because, but it becomes one when user can invest time to produce data/documents.

As global answer to ticket, I will list here all kinds of modification required to update charactersheet.

# Minimal update

If you want your charactersheet to be displayed exactly as it was on Rolisteam v1.8. The minimal change is for you.

## Minimal update: First change - import

in qml code, replace `import "qrc:/resources/qml/" ` with "import Rolisteam 1.0"

It could be enough to make the sheet ready to use. But I recommand few other steps.

## Minimal update: Second change - Sheet file

You may find useful to add those fields inside sheet at the first level in the json file.

* "additionnalBottomCode"
* "additionnalHeadCode"
* "additionnalImport"

Values can be empty string. For the two next key, values are required:

* uuid (value must be the first part of image key).
* pageCount (value must be the number of page of this sheet)


With those changes, RCSE will be able to open this sheet and display the qml.
Be careful, don't regerate the sheet or it may changed the visual aspect. 



# Complete Update:

To change the sheet to use rolisteam v1.9 sdk, it is easy. The visual aspect could be diffirent. The checkbox for example have been improved.

In order to update to SDK 1.9, you must do all change from *minimal update* and few others.

## Complete Update:  First change - import

in qml code, replace `import "qrc:/resources/qml/" ` with "import Rolisteam 1.1"
Pay attention, complete update means **1.1**, minimal update the version is **1.0**

Don't forget to do the second step of the minimal update.

## Complete Update: Second change - fields

Each time, a field has textColor and color properties, those properties must be renamed as follow:

textColor => color
color => backgroundColor

_________________  


Please find a python script atteched to this message. This script is able to update charactersheet from v1.8 to v1.9 with the complete update method.
The minimal change mode has not been tested yet.

Example:
```
./convertSheet.py -i Legend_of_five_ring_NPC/L5R_NPC.rcs -o NPC_L5R_fixed.rcs -c
```

-i input 
-o ouput 
-c => complete update mode.

I would like to test several sheets with this script to be sure it covers all kinds of sheet.
When it is ready, I will port the code into web language.

[convertSheet.zip](https://github.com/Rolisteam/rolisteam/files/3722640/convertSheet.zip)
